src/serialize.rs

```diff
@@ -312,7 +315,13 @@ impl EGraph {
                 let op = if sort.is_container_sort() {
                     sort.serialized_name(value).to_string()
                 } else {
-                    sort.make_expr(self, *value).1.to_string()
+                    let mut termdag = TermDag::default();
+                    let extractor = Extractor::new(self, &mut termdag);
+                    let (_, term) = sort
+                            .extract_term(self, *value, &extractor, &mut termdag)
+                            .expect("Extraction should be successful since extractor has been fully initialized");
+
+                    termdag.term_to_expr(&term).to_string()
```

Comment:
Thanks! I think this makes sense inlined here for now. It sets us up in case we want to refactor the extraction to batch extractions or something in the future if we want.

Reply:
Could you elaborate a little more about the change here? the make_expr is changed to extract_term, which returns an additional Optiion wrapper. The only case it returns None is when the extractor is not initialized. Is my understanding correct? When would extract_term return None?

Reply:
> Could you elaborate a little more about the change here? the make_expr is changed to extract_term, which returns an additional Optiion wrapper. The only case it returns None is when the extractor is not initialized. Is my understanding correct? When would extract_term return None?

`extract_term` would return None if `find_best` returns `None`, i.e. when one of the operands lowering hasn't been computed yet